test/arrays/errors/domainTypeAsDomainExpr.good

```diff
@@ -1,2 +1,3 @@
+domainTypeAsDomainExpr.chpl:11: warning: push_back is deprecated - please use List.extend
```

Comment:
@dlongnecke-cray: For this test, the use of push_back is completely independent of the purpose of the test (I think it was inherited from some code a user sent in).  Would you mind editing domainTypeAsDomainExpr.chpl  as follows:

```diff
-var x: [0..-1] owned Child;
-x.push_back(new owned Child("C"));
+var x: [0..0] owned Child = new owned Child("C");
```

(otherwise, I can do this in a separate PR post-merge).

(Pro tip: Saying the above reminds me of another strategy for tackling big changes to the test system like this:  check in an ability for developers to opt into the new behavior — like the config param you've added — check that in, list the tests that need to be dealt with, find owners for each (as I did here w.r.t. the reviews), and then have people rewrite the tests themselves to spread out the effort; then once everyone's done their bit, flip the config param bit on master.  This tends to work best when deadlines are further away and people are feeling most calm).

Reply:
Sounds good, I will admit I was a bit confused about that code at the time.